test/martha/martha_v3.test.js

```diff
@@ -351,321 +552,449 @@ test.serial('martha_v3 should return the underlying status if Data Object resolu
         t.is(
-            getJsonFromApiStub.firstCall.args[0],
+            getJsonFromApiStub.getCall(0).args[0],
             'https://dataguids.org/ga4gh/dos/v1/dataobjects/a41b0c4f-ebfb-4277-a941-507340dea85d',
         );
-        t.falsy(getJsonFromApiStub.firstCall.args[1]); // no auth passed
-        const requestedBondUrl = getJsonFromApiStub.secondCall.args[0];
-        const matches = requestedBondUrl.match(bondRegEx);
-        t.truthy(matches, 'Bond URL called does not match Bond URL regular expression');
-        const expectedProvider = 'dcf-fence';
-        const actualProvider = matches[2];
-        t.is(actualProvider, expectedProvider);
+        t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
+        const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
+        const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
+        t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
+        const expectedSAKeyProvider = 'dcf-fence';
+        const actualSAKeyProvider = saKeyUrlMatches[2];
+        t.is(actualSAKeyProvider, expectedSAKeyProvider);
     }
 );
 
 test.serial('martha_v3 does not call Bond or return SA key when the host url is for jade data repo', async (t) => {
-    getJsonFromApiStub.onFirstCall().resolves(jadeDrsResponse);
+    getJsonFromApiStub.onCall(0).resolves(jadeDrsResponse);
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
-    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    sinon.assert.callCount(getJsonFromApiStub, 1); // Bond was not called to get SA key
     t.deepEqual({ ...result }, jadeDrsMarthaResult);
     t.falsy(result.googleServiceAccount);
-    t.is(response.statusCode, 200);
     t.is(
-        getJsonFromApiStub.firstCall.args[0],
+        getJsonFromApiStub.getCall(0).args[0],
         'https://jade.datarepo-dev.broadinstitute.org/ga4gh/drs/v1/objects/abc',
     );
-    t.is(getJsonFromApiStub.firstCall.args[1], 'bearer abc123');
+    t.is(getJsonFromApiStub.getCall(0).args[1], 'bearer abc123');
 });
 
 test.serial('martha_v3 parses Gen3 CRDC response correctly', async (t) => {
-    getJsonFromApiStub.onFirstCall().resolves(gen3CrdcResponse);
+    const drsAccessUrlResponse = mockGcsAccessUrl(gen3CrdcResponse.access_methods[0].access_url.url);
+    getJsonFromApiStub.onCall(0).resolves(gen3CrdcResponse);
+    getJsonFromApiStub.onCall(1).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.onCall(2).resolves(drsAccessUrlResponse);
+    getJsonFromApiStub.onCall(3).resolves(googleSAKeyObject);
     const response = mockResponse();
     await marthaV3(
         mockRequest({ body: { 'url': `dos://${config.HOST_CRDC_STAGING}/206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc` } }),
```

Comment:
Should this be `drs://`?

Reply:
Despite attempts to remove DOS from the internet, it did exist and still does.